src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxInputNode.cs

```diff
@@ -101,7 +101,11 @@ public void VisitTree(Lazy<SyntaxNode> root, EntryState state, SemanticModel? mo
                         var value = new GeneratorSyntaxContext(node, model);
                         var transformed = _owner._transformFunc(value, cancellationToken);
 
-                        if (state == EntryState.Added || !_transformTable.TryModifyEntry(transformed, _owner._comparer, stopwatch.Elapsed, noInputStepsStepInfo, state))
+                        // The SemanticModel we provide to GeneratorSyntaxContext is never guaranteed to be the same between runs,
+                        // so we never consider the input to the transform as cached.
```

Comment:
Should the caller of `VisitTree()` be responsible for passing the correct `EntryState` in `state` instead? (Why is the incoming `state` ever set to `Cached` if that value is ignored?)

Reply:
The passed-in `EntryState` value can be cached for the "filter" step as it represents the "what is the state of the syntax tree represented by `root`".

The `GeneratorSyntaxContext` basically functions as a `Combine  node of a `SyntaxNode` and the `Compilation`, but since we already handle the "Compilation is cached" situation previously, we don't need to handle it here in the same way. We can assume that if we get here, that the Compilation changed, so we need to make the "input state" for the transform `Modified` as one of the inputs (the semantic model/compilation) has changed.